### PR TITLE
Connecting To MongoDB and building API for first AddNewPost

### LINKED ISSUE #1: Add a New Post
Add a New Post: Allows a user to add a new post to the database.

### PR DESCRIPTION
**Pull request by mevo0108**
closes #1

### CODE REVIEW

File: app_ex1.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+
```
Comment (taljacob2) [inline app_ex1.js:1]: Please rename this file to `app.js`

File: controllers/post.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+const createPost = async (req, res) => {
+    const post = req.body;
+    try{
+        const newPost = await postModel.create(post);
+        res.status(201).send(newPost);
+
+    }catch(error){
+        res.status(400).send(error);
```
Comment (taljacob2) [inline controllers/post.js:27]: Should we change this `send(error)` to `send(error.message)` to match line 15?

File: app_ex1.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+const express = require('express');
+const router = express.Router();
+const dotenv = require("dotenv").config();
+const app = express();
```
Comment (taljacob2) [inline app_ex1.js:5]: Please accept this code, so the `.env` file would be able to expand a variable by other variables.

```suggestion
const dotenv = require("dotenv");
const dotenvExpand = require("dotenv-expand");
dotenvExpand.expand(dotenv.config());
```

Take note that you should install the `dotenv-expand` package with `npm i dotenv-expand`, and commit the updated `package.json`.

Then, for example you could use in your `.env`:

```sh
USERNAME=John
HELLO_MESSAGE="Hello ${USERNAME}"
```

File: models/posts_model.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    title: {
+        type: String,
+        required: true
+    },
+    content: String,
+    owner: {
+        type: String,
+        required: true
+    },
+});
```
Comment (taljacob2) [inline models/posts_model.js:15]: Please accept this code, so the post document wouldn't have the generated `__v` property.

```suggestion
}, {
    versionKey: false,
});
```

Example Before:
```json
{
  "_id": "6744b4dea4ce2442aeddf377",
  "title": "Post 1",
  "content": "This is my first post",
  "owner": "Mevorah",
  "__v": 0
}
```

Example After:
```json
{
  "title": "Post 1",
  "content": "This is my first post",
  "owner": "Mevorah",
  "_id": "6744b55259e23afd32c6e2ac"
}
```

File: controllers/post.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+const postModel = require('../models/posts_model');
```
Comment (taljacob2) [inline controllers/post.js:1]: Please rename this file to `posts_controller.js`

File: app_ex1.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+const port = process.env.PORT;
```
Comment (taljacob2) [inline app_ex1.js:6]: Please add a new file called `.env.template` in the root directory of the project with the following content:

```sh
PORT=<PORT>
DB_CONNECTION=<DB_CONNECTION>
```

So other people would know that they are required to create an `.env` file with these variables to run the application.

**Review by taljacob2 — CHANGES_REQUESTED** [review]
Overall great job! 💯 

A few modifications are required for final approval.
Please make them, and we're done!

File: app_ex1.js (hunk 1 above)
Comment (mevo0108) [reply app_ex1.js:1]: no PB

File: app_ex1.js (hunk 3 above)
Comment (mevo0108) [reply app_ex1.js]: no problem i would be happy for more info about it :)

File: controllers/post.js (hunk 5 above)
Comment (mevo0108) [reply controllers/post.js]: done

**Review by mevo0108 — COMMENTED** [review]
finish to pass on all the review .
waiting for your approved to add the new file that you ask me for (.env.tamplate) on a new assignement (TODO)

**Review by mevo0108 — COMMENTED** [review]
@taljacob2

File: routes/posts_routes.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+
+const express = require('express');
+const router = express.Router();
+const Post = require('../controllers/post');
+
```
Comment (taljacob2) [inline routes/posts_routes.js:5]: Please fix this typo of renaming posts controller:

```suggestion
const Post = require('../controllers/posts_controller');
```

**Review by taljacob2 — APPROVED** [review]
Approved, need to add `.env.template` at #9

File: app_ex1.js (hunk 6 above)
Comment (taljacob2) [reply app_ex1.js]: Resolved in #9

**Pull request merged**